fox_consumer/consumer_db_support.py

```diff
@@ -99,31 +96,39 @@ def is_item_exist(self, item):
         :return: True/False (exist/not)
         """
         sql_query = "SELECT * FROM fox_items WHERE item_img_id = '%s';" % item.item_img_id
-        item_exist = self.execute_and_fetch_sql_query(sql_query)
-        return True if item_exist else False
+        item_exist = self.is_exist_execute_and_fetch_query(sql_query)
+        return item_exist
 
     def restart_connection(self):
         """
         restart connection (while the program is running)
         """
-        self.close_connections()
+        self.close_connection()
         self.connect_to_postgres()
 
-    def handle_postgres_general_exception(self, e, location):
+    def handle_postgres_general_exception(self, e):
         """
         handle general exception for postgres
         :param e: current exception
         :param location: source method the exception has thrown
         """
-        if e.pgcode == '42P01':
 
-            
-            print "PostgresWrapper - execute_and_commit - exception - 'undefined_table'"
-            print "Reconnecting and creating table..."
+        if isinstance(e, psycopg2.OperationalError):
+            # do not raise - fix it locally
+            print "PostgresWrapper - OperationalError exception:"
             self.restart_connection()
-            self.check_and_create_fox_table()
-            print "Info - last query did not applied !"
-        else:
-            print "PostgresWrapper - execute_and_commit - exception:"
-            print e
+            print "Error - last query did not applied !"
+        elif isinstance(e, psycopg2.ProgrammingError):
+            if e.pgcode == '42P01':
+                # do not raise - fix it locally - create a new table (need to restart the connection!)
+                print "PostgresWrapper - exception - 'unidentified_table'"
+                print "Reconnecting and creating table..."
+                self.restart_connection()
+                self.check_and_create_fox_table()
+                print "Error - last query did not applied !"
+            else:
+                print "PostgresWrapper - ProgrammingError - exception:"
+                raise
+        elif isinstance(e, psycopg2.Error):
+            print "PostgresWrapper - Error - exception:"
             raise
```

Comment:
if the PostgresWrapper cannot identify the exception it will raise it to the consumer and the consumer will raise it to the main runner and it will close the connection.